### PR TITLE
Add TLS option to HTTP transport

### PR DESCRIPTION
**Pull request by vitus133**
This commit adds support for HTTPS transport

/assign @aneeshkp
/cc @jzding @nishant-parekh

### CODE REVIEW

File: pkg/protocol/http/http.go [hunk 1]
```diff
@@ -76,13 +77,14 @@ type Server struct {
 }
 
 // InitServer initialize http configurations
-func InitServer(serviceName string, port int, storePath string, dataIn <-chan *channel.DataChan,
+func InitServer(serviceName string, port int, tls *tls.Config, storePath string, dataIn <-chan *channel.DataChan,
 	dataOut chan<- *channel.DataChan, closeCh <-chan struct{},
 	onStatusReceiveOverrideFn func(e cloudevents.Event, dataChan *channel.DataChan) error,
-	processEventFn func(e interface{}) error) (*Server, error) {
+	processEventFn func(e interface{}) error, options ...bool) (*Server, error) {
```
Comment (aneeshkp) [inline pkg/protocol/http/http.go:83]: what is this processevent  fn returning as options?

File: pkg/protocol/http/http.go [hunk 2]
```diff
@@ -22,7 +23,6 @@ import (
 
 	cloudevents "github.com/cloudevents/sdk-go/v2"
 	httpClient "github.com/cloudevents/sdk-go/v2/client"
-	cehttp "github.com/cloudevents/sdk-go/v2/protocol/http"
 	httpP "github.com/cloudevents/sdk-go/v2/protocol/http"
```
Comment (aneeshkp) [inline pkg/protocol/http/http.go:26]: this is cloudevents http protocol used , will it help if we keep it  as cehttp

Comment (vitus133) [reply pkg/protocol/http/http.go:26]: Can I remove the httpP then? Otherwise it complains about duplicate import

Comment (aneeshkp) [reply pkg/protocol/http/http.go]: yes lets have it as cnehttp

**Comment by vitus133** [comment]
Another attempt to fix lint errors

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**